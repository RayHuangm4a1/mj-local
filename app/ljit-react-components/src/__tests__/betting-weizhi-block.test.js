import React from 'react';
import { shallow, mount, } from 'enzyme';
import BettingWeizhiBlock from '../components/betting-weizhi-block';

const { SOLID, HOLLOW, } = BettingWeizhiBlock.CheckboxStyleEnum;

describe('Betting weizhi block', () => {
	it('handle default props', () => {
		const {
			options,
			description,
			onPressCheckbox,
			checkboxStyle,
		} = BettingWeizhiBlock.defaultProps;

		expect(options).toMatchObject([]);
		expect(description).toBe('');
		expect(checkboxStyle).toBe(SOLID);
		expect(onPressCheckbox).toBeDefined();
		expect(onPressCheckbox).toBeInstanceOf(Function);

	});

	it('should renders correctly', () => {
		const wrapper = shallow(
			<BettingWeizhiBlock
				className="mock-class-name"
				options={[
					{
						name: '萬',
						isSelected: true,
					},
					{
						name: '千',
						isSelected: false,
					},
					{
						name: '百',
						isSelected: false,
					},
					{
						name: '十',
						isSelected: false,
					},
					{
						name: '個',
						isSelected: false,
					},
				]}
				description="您选择了 2 个位置，系统自动根据位置组合成 1 个方案。"
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-betting-weizhi-block', () => {
		const wrapper = shallow(<BettingWeizhiBlock/>);

		expect(wrapper.hasClass('ljit-betting-weizhi-block')).toEqual(true);
	});

	it('should be selectable by class ljit-betting-weizhi-block--solid', () => {
		const wrapper = shallow(<BettingWeizhiBlock/>);

		expect(wrapper.hasClass('ljit-betting-weizhi-block--solid')).toEqual(true);
	});

	it('should be selectable by class ljit-betting-weizhi-block--hollow', () => {
		const wrapper = shallow(<BettingWeizhiBlock checkboxStyle={HOLLOW}/>);

		expect(wrapper.hasClass('ljit-betting-weizhi-block--hollow')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<BettingWeizhiBlock className={className}/>);

		expect(wrapper.hasClass(className)).toEqual(true);
	});
	it('should handle onPressCheckbox event', () => {
		const onPressCheckbox = jest.fn();
		const wrapper = mount(
			<BettingWeizhiBlock
				onPressCheckbox={onPressCheckbox}
			/>
		);

		wrapper.props().onPressCheckbox();
		expect(onPressCheckbox).toHaveBeenCalledTimes(1);
		expect(onPressCheckbox).toHaveBeenCalled();
	});

	it('should mount in a full DOM', () => {
		const options = [
			{
				name: '萬',
				isSelected: true,
			},
			{
				name: '千',
				isSelected: false,
			},
			{
				name: '百',
				isSelected: false,
			},
			{
				name: '十',
				isSelected: false,
			},
			{
				name: '個',
				isSelected: false,
			},
		];
		const className = 'mock-class';
		const description = '您选择了 2 个位置，系统自动根据位置组合成 1 个方案。';
		const onPressCheckbox = () => {};
		const wrapper = mount(
			<BettingWeizhiBlock
				options={options}
				className={className}
				description={description}
				onPressCheckbox={onPressCheckbox}
			/>
		);

		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().options).toMatchObject(options);
		expect(wrapper.props().description).toEqual(description);
		expect(wrapper.props().onPressCheckbox).toEqual(onPressCheckbox);
	});
});

describe('Betting weizhi block CheckboxStyleEnum ', () => {
	it('should contains solid property', () => {
		const typeName = 'solid';
		const formatType = 'SOLID';

		expect(BettingWeizhiBlock.CheckboxStyleEnum[formatType]).toEqual(typeName);
	});

	it('should contains hollow property', () => {
		const typeName = 'hollow';
		const formatType = 'HOLLOW';

		expect(BettingWeizhiBlock.CheckboxStyleEnum[formatType]).toEqual(typeName);
	});
});