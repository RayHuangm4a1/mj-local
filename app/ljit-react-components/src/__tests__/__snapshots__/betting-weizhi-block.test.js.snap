// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Betting weizhi block should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BettingWeizhiBlock
    checkboxStyle="solid"
    className="mock-class-name"
    description="您选择了 2 个位置，系统自动根据位置组合成 1 个方案。"
    onPressCheckbox={[Function]}
    options={
      Array [
        Object {
          "isSelected": true,
          "name": "萬",
        },
        Object {
          "isSelected": false,
          "name": "千",
        },
        Object {
          "isSelected": false,
          "name": "百",
        },
        Object {
          "isSelected": false,
          "name": "十",
        },
        Object {
          "isSelected": false,
          "name": "個",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <CheckBox
            autoFocus={false}
            className=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value={true}
          >
            萬
          </CheckBox>,
          <CheckBox
            autoFocus={false}
            className=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value={false}
          >
            千
          </CheckBox>,
          <CheckBox
            autoFocus={false}
            className=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value={false}
          >
            百
          </CheckBox>,
          <CheckBox
            autoFocus={false}
            className=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value={false}
          >
            十
          </CheckBox>,
          <CheckBox
            autoFocus={false}
            className=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value={false}
          >
            個
          </CheckBox>,
        ],
        <div
          className="ljit-betting-weizhi-block__description"
        >
          <Icon
            color="primary"
            isTranslucent={false}
            size="small"
            spin={false}
            style={Object {}}
            type="info"
          />
          <span>
            您选择了 2 个位置，系统自动根据位置组合成 1 个方案。
          </span>
        </div>,
      ],
      "className": "ljit-betting-weizhi-block ljit-betting-weizhi-block--solid mock-class-name",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "萬",
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "children": "萬",
          "className": "",
          "disabled": false,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "value": true,
        },
        "ref": null,
        "rendered": "萬",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "千",
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "children": "千",
          "className": "",
          "disabled": false,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "value": false,
        },
        "ref": null,
        "rendered": "千",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "百",
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "children": "百",
          "className": "",
          "disabled": false,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "value": false,
        },
        "ref": null,
        "rendered": "百",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "十",
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "children": "十",
          "className": "",
          "disabled": false,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "value": false,
        },
        "ref": null,
        "rendered": "十",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "個",
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "children": "個",
          "className": "",
          "disabled": false,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "value": false,
        },
        "ref": null,
        "rendered": "個",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              color="primary"
              isTranslucent={false}
              size="small"
              spin={false}
              style={Object {}}
              type="info"
            />,
            <span>
              您选择了 2 个位置，系统自动根据位置组合成 1 个方案。
            </span>,
          ],
          "className": "ljit-betting-weizhi-block__description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "primary",
              "isTranslucent": false,
              "size": "small",
              "spin": false,
              "style": Object {},
              "type": "info",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "您选择了 2 个位置，系统自动根据位置组合成 1 个方案。",
            },
            "ref": null,
            "rendered": "您选择了 2 个位置，系统自动根据位置组合成 1 个方案。",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <CheckBox
              autoFocus={false}
              className=""
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              value={true}
            >
              萬
            </CheckBox>,
            <CheckBox
              autoFocus={false}
              className=""
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              value={false}
            >
              千
            </CheckBox>,
            <CheckBox
              autoFocus={false}
              className=""
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              value={false}
            >
              百
            </CheckBox>,
            <CheckBox
              autoFocus={false}
              className=""
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              value={false}
            >
              十
            </CheckBox>,
            <CheckBox
              autoFocus={false}
              className=""
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              value={false}
            >
              個
            </CheckBox>,
          ],
          <div
            className="ljit-betting-weizhi-block__description"
          >
            <Icon
              color="primary"
              isTranslucent={false}
              size="small"
              spin={false}
              style={Object {}}
              type="info"
            />
            <span>
              您选择了 2 个位置，系统自动根据位置组合成 1 个方案。
            </span>
          </div>,
        ],
        "className": "ljit-betting-weizhi-block ljit-betting-weizhi-block--solid mock-class-name",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "萬",
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "children": "萬",
            "className": "",
            "disabled": false,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "value": true,
          },
          "ref": null,
          "rendered": "萬",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "千",
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "children": "千",
            "className": "",
            "disabled": false,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "value": false,
          },
          "ref": null,
          "rendered": "千",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "百",
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "children": "百",
            "className": "",
            "disabled": false,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "value": false,
          },
          "ref": null,
          "rendered": "百",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "十",
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "children": "十",
            "className": "",
            "disabled": false,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "value": false,
          },
          "ref": null,
          "rendered": "十",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "個",
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "children": "個",
            "className": "",
            "disabled": false,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "value": false,
          },
          "ref": null,
          "rendered": "個",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                color="primary"
                isTranslucent={false}
                size="small"
                spin={false}
                style={Object {}}
                type="info"
              />,
              <span>
                您选择了 2 个位置，系统自动根据位置组合成 1 个方案。
              </span>,
            ],
            "className": "ljit-betting-weizhi-block__description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "primary",
                "isTranslucent": false,
                "size": "small",
                "spin": false,
                "style": Object {},
                "type": "info",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "您选择了 2 个位置，系统自动根据位置组合成 1 个方案。",
              },
              "ref": null,
              "rendered": "您选择了 2 个位置，系统自动根据位置组合成 1 个方案。",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
